site/documentation/content/api/event/index.md

```diff
@@ -130,3 +132,88 @@ The example below might be used by the MQTT adapter to indicate that a connectio
   }
 }
 ~~~
+
+
+## Event API for Kafka
+
+Hono can alternatively be configured to publish event messages to an Apache Kafka&reg; cluster instead of an AMQP Messaging Network. The Event API for Kafka is defined by means of Kafka messages.
+
+### Southbound Operations
+
+The following operation can be used by *Protocol Adapters* to send event messages received from devices to downstream consumers like *Business Applications*.
+
+#### Produce Event
+
+The protocol adapter writes messages to the tenant-specific topic `event-${tenant_id}` where `${tenant_id}` is the ID of the tenant that the client wants to upload event messages for.
```

Comment:
The `/` is not allowed in Kafka topic names. The only valid non-alphanumerical characters are `.`, `_`, and `-`. Underscore and period could collide in metric systems, so I decided to use the safe character.

Reply:
Tenant IDs that are based on UUIDs will contain several `-` characters. Will this be a problem, e.g. with distinguishing topic names?

Reply:
If we strictly adhere to the schema that a topic starts with the prefix consisting of endpoint and `-` we can always only consider the first `-`. The other two characters are also not prohibited in tenant names, right?

Reply:
no, they are not. But in the default configuration the tenant ID pattern excludes `/` ...
As long as the topic names consist of two segments only, we might get away with considering the first `-` as the delimiter. However, with the C&C discussions going on, I am not sure if we can safely assume that it will always only be two segments ...
It would be safer to come up with another idea, e.g. URI escaping the `-` in path segments or something like that.

Reply:
Before having the problem of topic names with more than two elements, we have the even more pressing issue here that tenant IDs may contain characters that are not allowed and prevent the topic to be created in the first place.

URI-encoding does not work because the `%` character is not allowed in topic names. To solve this we could invent our own encoding logic. I am hesitating to do this because backend applications would be required to do this as well, which might be error-prone.

A second option is to restrict tenant names (I would suggest to the characters used in UUIDs). Even if we don't want to change the pattern that validates tenant IDs with a minor version of Hono, it could be documented that using Kafka requires migrating tenant IDs to this restricted character set (plus starting, to create only safe tenant IDs in our device registry code).

A third option would be to not use the tenant ID in the topic name, but a new, safe identifier. This could be stored in the tenant configuration. The drawback is that backend applications could no longer derive it from the tenant ID but need somehow to get the topic name.

Reply:
FMPOV option 2 should be ok as we are not restricting any existing functionality but add new functionality here ...

Reply:
I just had a look at the tenant ID pattern. This is exactly the same character set that Kafka allows for topic names: https://github.com/eclipse/hono/blob/41ad9e6f73b8bb1cd1dd6742a19c6817b6aa9d94/core/src/main/java/org/eclipse/hono/util/RegistryManagementConstants.java#L343

So, we **do not** have the problem of invalid characters in current tenant IDs.

We could continue with the initial approach of static prefixes followed by the tenant ID. From my understanding, even for C&C, there should be no need for topics with more (dynamic) elements. 

Alternatively, if we do not feel safe with this approach, I would deprecate the current pattern and add a new one that does not include the dot ('.'). The dot would then be Hono's the delimiter character for topic names. I would still allow '_' because it is used in "DEFAULT_TENANT" which is widely used, even in external resources like blog articles (and of course '-' because of the UUIDs).

@sophokles73 WDYT?
@calohmn, do you expect topic names for C&C to contain multiple elements that would require to have a delimiter character?

Reply:
Concerning C&C, the envisioned approaches only use topics with a fixed prefix followed by the tenant ID or some other ID. So, for these, no delimiter would be needed.

Reply:
My proposal for topic names is: 
`hono.${endpoint}.${tenant_id}`.
1. Namespacing (i.e. starting with `hono`) is a good practice and might come in handy later on (e.g. Kafka supports subscriptions on multiple topics with regular expressions.
2. In contrast to my first proposal, I would not use `-` as a delimiter to make it more readable with UUIDs as tenant IDs. Since the `_` is already contained in the famous "DEFAULT_TENANT" and in the endpoint `command_response", the `.` is my first choice.

@sophokles73 @calohmn WDYT?

Reply:
Based on https://stackoverflow.com/questions/37062904/what-are-apache-kafka-topic-name-limitations it seems that we should use *either* `.` or `_` but not both. Given that we already have the `DEFALUT_TENANT` name, it looks like we are advised against using `.` as well then.

How about using `__` (two underscores) as the delimiter instead?

Reply:
> we should use _either_ `.` or `_` but not both

That is right. The reason is that they could collide in metrics. Therefore the Kafka CLI always prints a warning if you create a topic that contains a  `.` or a `_`. In practice, there would be no collision with our naming schema.

> How about using `__` (two underscores) as the delimiter instead?

I like the creative thinking here, but the drawback is that two underscores are hard to read. Depending on the font they are displayed as a single long underscore (obviously we need to explain at least here on Github what it is).

Reply:
I think it would be good if we could come to a final decision regarding the structure of topic names soon. FMPOV, we can't complete the API specification or implementation before that. 

My proposal would be the following: 
1. we start with using topic names with a static prefix and a dynamic second element. Then we do not need a reserved delimiter.
2. we separate the two elements (and possible sub-elements within the prefix) with a defined _separator_, which is only for human readability and not used as a _technical delimiter_ for parsing the name.
3. we take the '.' as the _separator_. Although we then have '.' and '_' in topic names, which in principle could lead to name conflicts in Kafka metrics, no conflicts will occur with our structure of the names.
4. if we realize at a later time that this naming scheme is no longer sufficient for us, we can still define a technical delimiter. For the TenantIds, '.' is a good choice for the reasons mentioned above, so that we don't have to change the character again. For other dynamic elements, we can't tell which characters they contain without knowing them anyway.
5. the static prefixes are additionally namespaced with `hono`.

The result will look like this:

`hono.${endpoint}.${tenant_id}`. Or, more concrete: `hono.event.${tenant_id}`, `hono.telemetry.${tenant_id}`, `hono.command.${tenant_id}`, `hono.command_response.${tenant_id}`.

WDYT?

Reply:
Looks ok FMPOV. What is the proposed pattern for tenant identifiers then?

Reply:
> Looks ok FMPOV. What is the proposed pattern for tenant identifiers then?

I think we could leave it as it is for now. If we consider everything after the fixed prefix as identifiers for the client, it doesn't matter what it contains. But if you feel more comfortable if we now deprecate '.' in the tenant identifiers, that's fine with me too.